source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2063,8 +2064,24 @@ namespace GridTools
     Assert(0 <= direction && direction < space_dim,
            ExcIndexRange(direction, 0, space_dim));
 
-    Assert(pairs1.size() == pairs2.size(),
-           ExcMessage("Unmatched faces on periodic boundaries"));
+#ifdef DEBUG
+    {
+      constexpr int dim      = CellIterator::AccessorType::dimension;
+      constexpr int spacedim = CellIterator::AccessorType::space_dimension;
+      // for parallel::fullydistributed::Triangulation there might be unmatched
+      // faces on periodic boundaries
```

Comment:
Is this a restriction you intend to lift eventually? This will prevent some algorithms from working.

Reply:
This is not a restriction. This might happen if a ghost coarse-grid cell has a periodic face and the periodic neighbor is not locally relevant.

Reply:
Hm, but *every* ghost neighbor is, by definition, a ghost cell and consequently locally relevant. No?

Reply:
No. 1D example with 8 cells partitioned into 2 domains and periodicity (at the top: ranks of the cells; at the bottom: labeling with ghost (g) and local (l)):
```
0 1 1 1 1 0 0 0
g l l l l g
```
The left ghost cell has periodic face on the left but its left neighbor (i.e. the most right cell) is not locally relevant.

Reply:
Ah, I see. Do you think you can put a summary of this (in particular, that it only works on the coarse mesh) into the comment here (and if there are copies of this comment, there as well)?